pkg/integrations/mssql/common/collector.go

```diff
@@ -0,0 +1,109 @@
+package common
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/burningalchemist/sql_exporter"
+	"github.com/burningalchemist/sql_exporter/config"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"gopkg.in/yaml.v2"
+)
+
+// Embedded config.CollectorConfig as yaml.
+// We do this so that we can unmarshal instead of creating a static instance in code;
+// This is because there is special unmarshal logic for the CollectorConfig
+// that sets some unexported fields.
+//
+//go:embed collector_config.yaml
+var collectorConfigBytes []byte
+
+// CollectorConfig is a config that can be used to construct a
+// sql_exporter.Target that scrapes metrics from an instance of mssql
+var CollectorConfig config.CollectorConfig
+
+// initialize static collector config
+func init() {
+	err := yaml.Unmarshal(collectorConfigBytes, &CollectorConfig)
+	if err != nil {
+		panic(fmt.Errorf("failed to unmarshal mssql integration collector config: %w", err))
+	}
+}
+
+// TargetCollectorAdapter adapts sql_exporter.Target to prometheus.Collector
+type TargetCollectorAdapter struct {
+	target sql_exporter.Target
+	logger log.Logger
+}
+
+// NewTargetCollectorAdapter creates a new TargetCollectorAdapter
+func NewTargetCollectorAdapter(t sql_exporter.Target, l log.Logger) TargetCollectorAdapter {
+	return TargetCollectorAdapter{
+		target: t,
+		logger: l,
+	}
+}
+
+// Collect calls the collect function of the underlying sql_exporter.Target, converting each
+// returned sql_exporter.Metric to a prometheus.Metric.
+func (t TargetCollectorAdapter) Collect(m chan<- prometheus.Metric) {
+	sqlMetrics := make(chan sql_exporter.Metric, 1000)
```

Comment:
Where is this value coming from? Is there any reason to have this bounded channel?

Reply:
It was arbitrary, good call. Made this channel unbuffered.